x/subspaces/types/models_test.go

```diff
@@ -583,3 +581,7 @@ func TestUserGroup_Update(t *testing.T) {
 		})
 	}
 }
+
+func TestGetTreasuryAddress(t *testing.T) {
+	require.Equal(t, "cosmos1cyjzgj9j7d2gdqk78pa0fgvfnlzradat97aek9", types.GetTreasuryAddress(1).String())
```

Comment:
I believe we should test this method more properly. Ideally, what we could do is: 
1. Generate a list of subspace ids
2. For each subspace id, generate the treasury address
3. Check that the generated addresses are all different from one another

This will make sure that we are not only always testing one case (`SubspaceID: 1`) but we are actually proving the generation method works for every subspace id

Reply:
Applied